### ISSUE TITLE
If a zone has two public IPs range in different VLANs, ACS doesn't create a second northbound NIC if needed

### ISSUE THREAD

**Issue by msinhore** [apache/cloudstack#3698]
##### ISSUE TYPE
~~~
VR
~~~

##### CLOUDSTACK VERSION
~~~
4.11.3
~~~

##### CONFIGURATION
- 2 public IPs ranges configured in the same zone using different VLANs.

##### SUMMARY
ACS is configured with 2 public IPs ranges in different VLANs. 
If a user creates a network (where the snat IP is from one of the public IP ranges) and request a new public IP address (ACS deliver from the another IP range), this second IP will be in the same VLAN of the first one, since the ACS doesn't create a second northbound (public) NIC in the VR. 

##### EXPECTED RESULTS

~~~
2 northbound NICs each connected in the right VLAN 
~~~

##### ACTUAL RESULTS
~~~
The second IP is configured on the same primary public IP NIC and doesn't have network access.
~~~

**Comment by weizhouapache** [comment]
@msinhore 
cloudstack will plug a new nic in virtual routers when the ip is in use.
For example, enable static nat, create load balancer rule or port forwarding rule.
If ip is Allocated but not in use, there is no nic for the ip/vlan in virtual routers.

This is not an issue I think.

**Comment by msinhore** [comment]
@weizhouapache 
When was added a new NIC, it was configured a static NAT too and looking inside the VR, the second NIC wasn't created. Considering this behavior, I presume it is a bug.

**Comment by weizhouapache** [comment]
@msinhore 
which hypervisor ?
isolated network or vpc ?
if kvm, linux bridge or openvswitch ?
is vm running ?

**Comment by msinhore** [comment]
@weizhouapache 
- XenServer 7.2
- VPC
- with VMs running.

**Comment by weizhouapache** [comment]
@msinhore 
I cannot reproduce the issue on kvm (linux bridge or ovs)
it might be caused by some code in xenserver plugin.

It might be related to issue #3179 
I have uploaded some commits to https://github.com/ustcweizhou/cloudstack/commits/4.14.0-vpc-vr-issues
not sure if it fixes your issue

**Comment by andrijapanicsb** [comment]
This works fine in 4.13 per my memory (from a few days ago). Can you test @msinhore ?

**Comment by andrijapanicsb** [comment]
I couldn't reproduce this on master:
ACS 4.14 (master)
XS 7.1

![image](https://user-images.githubusercontent.com/45762285/71767230-47dbff80-2f0a-11ea-8ba3-abfbb9f8f0dc.png)


![image](https://user-images.githubusercontent.com/45762285/71767210-2ed34e80-2f0a-11ea-9f34-e858f95834f8.png)

@msinhore let me know if it's possible to test master or 4.13 - otherwise, I would close this issue as not applicable any more (I assume it was fixed in the meantime from 4.11 until now)

**Comment by msinhore** [comment]
@andrijapanicsb 
I am closing this issue,  as you applied the needed tests and it isn't reproduced anymore.
